Upgrade JSON-FG, refactor support for profiles, add additional profiles

cportele opened the pull request:
### Pull request checklist

-   [x] Added/updated unit tests
-   [x] Updated documentation
-   [x] All checks are passing

### Changes introduced by this PR

Closes #1397.
Closes #521.
Closes #1419.
Closes #1420.
Closes #1421.

Depends on https://github.com/ldproxy/xtraplatform-spatial/pull/361.

As part of the changes to JSON-FG, including support for embedding in all writers, the logic in the `GeoJSONWriter` classes has been updated. The stack of the features (root feature and embedded features) is now in the feature transformation context. The properties are always now buffered until all other applicable members (except links) have been written.

There are breaking changes to some existing API configurations:

- APIs that support the JSON-FG+ profile (previously: GeoJSON compatibility mode) now have to set the role `PRIMARY_GEOMETRY` to the geometry property for the "place" member. For the fallback geometry in the "geometry" member, a separate geometry property with the role `SECONDARY_GEOMETRY` has to be specified (it can use the same source path / geometry).

JSON-FG v0.3 also includes breaking changes to existing APIs based on changes in the candidate standard, including:

- There is no longer a plan to register a JSON-FG media type. The GeoJSON media type is now used and three profiles control the details of the representation.
- CURIEs are no longer supported.
- If "place" and "time" have no value, they are omitted instead of writing a `null` value.

Note: Tested with the "embed", "crud", "gvd" and "demo" configurations.

cportele commented:
I have updated the HTML schema view to include additional information from the JSON Schema (role, codelist, referenced collection) plus some fixes.

I noticed still one open issue in the schemas. The order of the properties is completely lost after the processing and properties appear in arbitrary order. This is quite confusing.

azahnen commented:
@cportele 
- I will add the PR Todo Checker that we already have in xtraplatform-spatial. I saw some TODOs during the review, especially the disabled tests might have to be addressed.
- I made various compare tests to check the GeoJSON is unchanged and JSON-FG only has the expected changes.
- I testet the feature profiles, still work as expected.
- I tested the schema profiles, there seem to be some issues:
  -  the schema still uses `x-ldproxy-codelistUri`, I think it should be `x-ogc-codelistUri`
  - `codelists-inline` has no effect, I see warnings in the log 
    `Codelist with ID 'cologne_lod2/RoofType' not found, returning schema without codelist information`
    http://localhost:7080/cologne_lod2/collections/building/schema?f=json&profile=codelists-inline

cportele commented:
@azahnen 

Thanks for the review. I have made the following changes:

- The problem with the codelist access in `cologne_lod2` has been fixed (the cause was that codelists in sub-folders was not handled properly).
- I have restored the "old" `featureType` configuration option, deprecated it, added `featureTypeV1` and a migration function to effectively use `featureTypeV1` only.
- I fixed the use of "x-ldproxy-codelistUri". I think this was still using "ldproxy" as at the time when I made the first changes, the codelist profile was not yet specified in Part 5.
- I have enabled the ignored unit test and updated it. I have also added a few more, but I have also removed some ignored tests that were really outdated. As GeoJSON and JSON-FG are key formats, we probably should add a follow-up task to add tests with more coverage (including embedded features, etc).
- The other TODOs were also reviewed (in general, these were not new).

azahnen reviewed (approved):
Looks good to me now.
For the TODOs you want to merge, you have to check the `Ignore` boxes. I think the bot will then resolve the conversations.
